tripal_chado/src/Plugin/TripalIdSpace/ChadoIdSpace.php

```diff
@@ -163,21 +171,35 @@ public function destroy(){
    * This function queries the `db` table of Chado to get the values
    * for the ID space.
    *
+   * @param bool $force_reload
+   *   If true the record will always be loaded from chado, if false the cache
+   *   will be used if it exists.
+   *
    * @return array
    *   An associative array containing the columns of the `db1 table
    *   of Chado or NULL if the db could not be found.
    */
-  protected function loadIdSpace() {
-
-    // Get the Chado `db` record for this ID space.
-    $query = $this->connection->select('1:db', 'db')
-      ->condition('db.name', $this->getName(), '=')
-      ->fields('db', ['name', 'url', 'urlprefix', 'description']);
-    $result = $query->execute();
-    if (!$result) {
-      return NULL;
+  protected function loadIdSpace(bool $force_reload = TRUE) {
+
+    if ($force_reload OR $this->chado_record === NULL) {
```

Comment:
```suggestion
    if ($force_reload or $this->chado_record === NULL) {
```
fussy, but in the other case of this test, the OR is lower case